SubGHz: Add 9 new protocols, fixes to existing protocols

xMasterX opened the pull request:
# What's new

Wow, 2 years of work in one PR 🥲 😸 

- Add new protocols:
  * Roger (by @xMasterX & @mishamyte)
  * Revers RB2(M) (by @xMasterX)
  * Feron (by @xMasterX & @Svaarich)
  * Marantec24 (by @xMasterX)
  * Hay21 (by @xMasterX)
  * Legrand (by @user890104)
  * GangQi (by @xMasterX & @mishamyte & @DoberBit & @Skorpionm)
  * Hollarm (by @xMasterX & @mishamyte & @Skorpionm)

- Improve existing protocols:
  * V2 Phoenix - Decrypt counter value and show on screen (by @xMasterX & @RocketGod-git)
  * Marantec - Display if CRC is valid or not, parse more data (by @xMasterX & @li0ard)
  * ScherKhan - Display more types of received signal (by @xMasterX)
  * Magellan - Better parsing and more types (by @wooferguy)

- Add new Keeloq keys:
  * IL-100(Smart) (thx Vitaly for RAWs)
  * Motorline (by @RocketGod-git)
  * Rosh (by @RocketGod-git)
  * Pecinin (by @RocketGod-git)
  * Rossi (by @RocketGod-git)
  * Merlin (by @RocketGod-git)
  * Steelmate (by @RocketGod-git)

- Fix existing protocols:
  * CAME TOP - Tune decoder to decode more original remotes (2025 year +)
  * Holtek HT12X - Tune decoder to fix false positivites
  * RAW - Fix typos and add extra check for sample duration
  * Alutech AT-4N - Fix name
  * Dooya - Fix typos in code
  * Fix typo in public api .h

# Verification 

- Try decoding RAWs from unit test folder

<details>

<summary>Check unit test passing</summary>

- Protocols
 - [x] Roger (by @xMasterX & @mishamyte)
 - [x] Revers RB2(M) (by @xMasterX)
 - [x] Feron (by @xMasterX & @Svaarich)
 - [x] Marantec24 (by @xMasterX)
 - [x] Hay21 (by @xMasterX)
 - [x] Legrand (by @user890104)
 - [x] GangQi (by @xMasterX & @mishamyte & @Drone1950 & @Skorpionm)
 - [x] Hollarm (by @xMasterX & @mishamyte & @Skorpionm)

</details>

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix
-

xMasterX committed:
Fix Typos

xMasterX committed:
Tune decoders

xMasterX committed:
Better parsing, show more data in existing protocols

xMasterX committed:
Add new protocols

xMasterX committed:
Update keeloqs

xMasterX committed:
Add unit tests & raws

htotoo commented:
Here are some Honeywell sec raw files:
[honey_subz.zip](https://github.com/user-attachments/files/21505073/honey_subz.zip)

zhovner commented:
Thanks a lot. Preparing merge

xMasterX committed:
Add honeywell unittest

xMasterX commented:
> Here are some Honeywell sec raw files: [honey_subz.zip](https://github.com/user-attachments/files/21505073/honey_subz.zip)

Thanks!

xMasterX committed:
Merge branch 'dev' into big_subghz

xMasterX committed:
Comment until better solution is found

Adding GAPs to be sent first to make signal better suitable for decoder (decoding from only one signal sample) does nothing, needs something else
TODO: Fix encoders?

hedger committed:
Merge branch 'dev' into big_subghz

hedger committed:
suppressed missing issue warning

hedger committed:
subghz: re-enabled failing encoder tests

xMasterX committed:
Merge remote-tracking branch 'origin/dev' into big_subghz

hedger commented:
Drafting until encoders/decoders work properly with unit tests

xMasterX committed:
Merge branch 'dev' into big_subghz

xMasterX committed:
Fix two?

3 left

xMasterX committed:
properly do gangqi and marantec for unit test and real use

xMasterX committed:
Merge branch 'dev' into big_subghz

xMasterX committed:
Merge branch 'dev' into big_subghz

xMasterX committed:
fix unit tests now

hedger committed:
Merge branch 'dev' into big_subghz

xMasterX committed:
fix possible memory leak

hedger committed:
Merge branch 'dev' into big_subghz

hedger committed:
Merge branch 'dev' into big_subghz

xMasterX committed:
reset decoder step too

hedger committed:
subghz: extra encoder safety; report random signal test results on failure

hedger committed:
unit_tests: subghz: renamed test file for consistency

hedger committed:
subghz: more explicit buffer position resets

xMasterX committed:
Fix gangqi samples

noproto commented:
Awesome work! :tada:
